app/assets/javascripts/controllers/project_roles_ctrl.js

```diff
@@ -0,0 +1,126 @@
+samson.controller('ProjectRolesCtrl', function ($scope, $element, $filter, projectRolesService, messageCenterService) {
+    $scope.project_role = {};
+    $scope.roles = [];
+
+    $scope.initModel = function () {
+        $scope.project_role = {
+            id: getUniqueId(),
+            user_id: getUserId(),
+            project_id: getProjectId(),
+            role_id: getRoleId()
+        };
+
+        loadProjectRoles();
+    };
+
+    $scope.$watch('project_role.role_id', function (new_role_value, old_role_value) {
+        if (new_role_value !== old_role_value) {
+            if (exists($scope.project_role)) {
+                return updateProjectRole($scope.project_role, new_role_value);
+            }
+            else {
+                return createProjectRole($scope.project_role, new_role_value);
+            }
+        }
+    });
+
+    function loadProjectRoles() {
+        projectRolesService.loadProjectRoles().then(
+            function (response) {
+                $scope.roles = response.data;
+            }
+        );
+    }
+
+    function exists(project_role) {
+        return project_role.id;
```

Comment:
This is a strange function anyway, I don't think it makes the code more readable then
```
if (project_role.id) { /* do stuff */ }
```